cli/api/commands/test.ts

```diff
@@ -83,6 +83,15 @@ async function runTest(
       const normalizedColumn = normalizeColumnName(column);
       const expectedValue = expectedResultRow[normalizedColumn];
       const actualValue = actualResultRow[normalizedColumn];
+      // Check if exactly one of the expected or actual values is null (XOR condition)
+      if ((expectedValue === null) !== (actualValue === null)) {
+        const expectedText = expectedValue === null ? 'null' : `"${expectedValue}"`;
+        const actualText = actualValue === null ? 'null' : `"${actualValue}"`;
+        rowMessages.push(
+          `For row ${i} and column "${column}": expected ${expectedText}, but saw ${actualText}.`
+        );
+        break;
+      }
```

Comment:
I much preferred your suggestion of

```ts
      // Null value check
      if (expectedValue === null && actualValue !== null) {
        rowMessages.push(
          `For row ${i} and column "${column}": expected null, but saw "${actualValue}".`
        );
        break;
      }
      if (expectedValue !== null && actualValue === null) {
        rowMessages.push(
          `For row ${i} and column "${column}": expected "${expectedValue}", but saw null.`
        );
        break;
      }
```

This is much clearer to follow the logic of!

Reply:
You're right, I also think that keeping it simple makes it easier to understand. I've committed.
https://github.com/dataform-co/dataform/pull/1812/commits/d40e265322ba45ec6236f52df0d3f9e7e8d2cd24

Reply:
@Ekrekr 
Is there a plan to merge this PR?